In [98]:
import pandas as pd

bo7raw = pd.read_csv('cdlHillCountsBO7.csv')
bo7raw

,Date,Venue,Team A,Team B,Map,Hill 1 A,Hill 1 B,Hill 2 A,Hill 2 B,Hill 3 A,...,Hill 9 A,Hill 9 B,Hill 10 A,Hill 10 B,Hill 11 A,Hill 11 B,Hill 12 A,Hill 12 B,Hill 13 A,Hill 13 B
0,12/5/2025,Online,Falcons,OpTic,Scar,0,40,24,59,42,...,154.0,245.0,154.0,250.0,NaN,NaN,NaN,NaN,NaN,NaN
1,12/5/2025,Online,Ravens,Breach,Colossus,19,4,49,23,58,...,153.0,228.0,153.0,250.0,NaN,NaN,NaN,NaN,NaN,NaN
2,12/5/2025,Online,Ravens,Breach,Exposure,23,6,34,38,40,...,174.0,219.0,203.0,242.0,214.0,250.0,NaN,NaN,NaN,NaN
3,12/5/2025,Online,Thieves,FaZe,Den,7,24,7,67,28,...,201.0,183.0,224.0,207.0,250.0,225.0,NaN,NaN,NaN,NaN
4,12/6/2025,Online,KOI,Heretics,Scar,29,0,80,0,131,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
109,2/15/2026,Online,M8s,Heretics,Den,39,3,75,9,108,...,226.0,172.0,250.0,172.0,NaN,NaN,NaN,NaN,NaN,NaN
110,2/15/2026,Online,Falcons,Breach,Exposure,6,24,20,38,58,...,180.0,197.0,237.0,197.0,250.0,204.0,NaN,NaN,NaN,NaN
111,2/15/2026,Online,Falcons,Breach,Colossus,13,25,41,49,42,...,149.0,207.0,189.0,207.0,213.0,221.0,217.0,250.0,NaN,NaN
112,2/15/2026,Online,Thieves,Ravens,Colossus,28,4,57,25,86,...,184.0,250.0,201.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN


Here we have a dataset that records the scores of matches in the Call of Duty League, specifically in the gamemode "Hardpoint". I want to explore with this where teams are performing the best, the worst, how they score their points, and other nuances of this fast paced game.

To start, the data will be transformed so that the points are shown by the amount earned each hill, rather than the total score at that point in the game.

In [99]:
timePerHill = bo7raw.copy()

timePerHill['Hill 13 A'] -= timePerHill['Hill 12 A']
timePerHill['Hill 13 B'] -= timePerHill['Hill 12 B']

timePerHill['Hill 12 A'] -= timePerHill['Hill 11 A']
timePerHill['Hill 12 B'] -= timePerHill['Hill 11 B']

timePerHill['Hill 11 A'] -= timePerHill['Hill 10 A']
timePerHill['Hill 11 B'] -= timePerHill['Hill 10 B']

timePerHill['Hill 10 A'] -= timePerHill['Hill 9 A']
timePerHill['Hill 10 B'] -= timePerHill['Hill 9 B']

timePerHill['Hill 9 A'] -= timePerHill['Hill 8 A']
timePerHill['Hill 9 B'] -= timePerHill['Hill 8 B']

timePerHill['Hill 8 A'] -= timePerHill['Hill 7 A']
timePerHill['Hill 8 B'] -= timePerHill['Hill 7 B']

timePerHill['Hill 7 A'] -= timePerHill['Hill 6 A']
timePerHill['Hill 7 B'] -= timePerHill['Hill 6 B']

timePerHill['Hill 6 A'] -= timePerHill['Hill 5 A']
timePerHill['Hill 6 B'] -= timePerHill['Hill 5 B']

timePerHill['Hill 5 A'] -= timePerHill['Hill 4 A']
timePerHill['Hill 5 B'] -= timePerHill['Hill 4 B']

timePerHill['Hill 4 A'] -= timePerHill['Hill 3 A']
timePerHill['Hill 4 B'] -= timePerHill['Hill 3 B']

timePerHill['Hill 3 A'] -= timePerHill['Hill 2 A']
timePerHill['Hill 3 B'] -= timePerHill['Hill 2 B']

timePerHill['Hill 2 A'] -= timePerHill['Hill 1 A']
timePerHill['Hill 2 B'] -= timePerHill['Hill 1 B']

timePerHill

,Date,Venue,Team A,Team B,Map,Hill 1 A,Hill 1 B,Hill 2 A,Hill 2 B,Hill 3 A,...,Hill 9 A,Hill 9 B,Hill 10 A,Hill 10 B,Hill 11 A,Hill 11 B,Hill 12 A,Hill 12 B,Hill 13 A,Hill 13 B
0,12/5/2025,Online,Falcons,OpTic,Scar,0,40,24,19,18,...,20.0,29.0,0.0,5.0,NaN,NaN,NaN,NaN,NaN,NaN
1,12/5/2025,Online,Ravens,Breach,Colossus,19,4,30,19,9,...,12.0,22.0,0.0,22.0,NaN,NaN,NaN,NaN,NaN,NaN
2,12/5/2025,Online,Ravens,Breach,Exposure,23,6,11,32,6,...,5.0,45.0,29.0,23.0,11.0,8.0,NaN,NaN,NaN,NaN
3,12/5/2025,Online,Thieves,FaZe,Den,7,24,0,43,21,...,27.0,12.0,23.0,24.0,26.0,18.0,NaN,NaN,NaN,NaN
4,12/6/2025,Online,KOI,Heretics,Scar,29,0,51,0,51,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
109,2/15/2026,Online,M8s,Heretics,Den,39,3,36,6,33,...,1.0,30.0,24.0,0.0,NaN,NaN,NaN,NaN,NaN,NaN
110,2/15/2026,Online,Falcons,Breach,Exposure,6,24,14,14,38,...,12.0,33.0,57.0,0.0,13.0,7.0,NaN,NaN,NaN,NaN
111,2/15/2026,Online,Falcons,Breach,Colossus,13,25,28,24,1,...,20.0,13.0,40.0,0.0,24.0,14.0,4.0,29.0,NaN,NaN
112,2/15/2026,Online,Thieves,Ravens,Colossus,28,4,29,21,29,...,25.0,21.0,17.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [100]:
timePerHill[timePerHill.columns[5:31]].max()

Hill 1 A      40.0
Hill 1 B      40.0
Hill 2 A      57.0
Hill 2 B      54.0
Hill 3 A      53.0
Hill 3 B      49.0
Hill 4 A      53.0
Hill 4 B      91.0
Hill 5 A      56.0
Hill 5 B      57.0
Hill 6 A      52.0
Hill 6 B      55.0
Hill 7 A      53.0
Hill 7 B      53.0
Hill 8 A      56.0
Hill 8 B      55.0
Hill 9 A      59.0
Hill 9 B      55.0
Hill 10 A    101.0
Hill 10 B     55.0
Hill 11 A     46.0
Hill 11 B     58.0
Hill 12 A     50.0
Hill 12 B     49.0
Hill 13 A     15.0
Hill 13 B     24.0
dtype: float64

Uh oh, 60 should be the maximum possible points for a particular hill, but here we have at least two instances that being broken. Let's have a look at these particular hills.

In [102]:
check4b = timePerHill.sort_values(by=['Hill 4 B'], ascending=False)

In [103]:
check4b

,Date,Venue,Team A,Team B,Map,Hill 1 A,Hill 1 B,Hill 2 A,Hill 2 B,Hill 3 A,...,Hill 9 A,Hill 9 B,Hill 10 A,Hill 10 B,Hill 11 A,Hill 11 B,Hill 12 A,Hill 12 B,Hill 13 A,Hill 13 B
112,2/15/2026,Online,Thieves,Ravens,Colossus,28,4,29,21,29,...,25.0,21.0,17.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN
58,1/23/2026,Online,OpTic,Ravens,Den,16,24,13,35,36,...,0.0,18.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
44,1/16/2026,Online,Falcons,M8s,Exposure,7,21,10,42,21,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
52,1/18/2026,Online,Cloud9,Breach,Colossus,4,27,53,0,25,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
77,1/30/2026,LAN,Ravens,Surge,Colossus,3,28,29,21,15,...,25.0,11.0,30.0,17.0,38.0,12.0,3.0,5.0,NaN,NaN
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
65,1/24/2026,Online,Thieves,Breach,Den,27,2,20,17,19,...,15.0,29.0,17.0,19.0,14.0,5.0,NaN,NaN,NaN,NaN
82,1/30/2026,LAN,FaZe,Breach,Den,24,22,41,10,11,...,28.0,19.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
15,12/12/2025,Online,Falcons,Ravens,Colossus,31,5,15,23,12,...,37.0,7.0,27.0,21.0,NaN,NaN,NaN,NaN,NaN,NaN
91,2/1/2026,LAN,FaZe,G2,Blackheart,31,5,28,13,17,...,28.0,14.0,13.0,23.0,31.0,9.0,0.0,2.0,NaN,NaN


Going to the February 15th match between the Thieves and Ravens does show an error in our data. The true value for "Hill 4 B" was missing and caused the anamoly seen above. Now lets do the same for 10 A

In [104]:
check10a = timePerHill.sort_values(by=['Hill 10 A'], ascending=False)
check10a

,Date,Venue,Team A,Team B,Map,Hill 1 A,Hill 1 B,Hill 2 A,Hill 2 B,Hill 3 A,...,Hill 9 A,Hill 9 B,Hill 10 A,Hill 10 B,Hill 11 A,Hill 11 B,Hill 12 A,Hill 12 B,Hill 13 A,Hill 13 B
100,2/13/2026,Online,Heretics,FaZe,Scar,23,4,43,1,23,...,-71.0,19.0,101.0,0.0,NaN,NaN,NaN,NaN,NaN,NaN
110,2/15/2026,Online,Falcons,Breach,Exposure,6,24,14,14,38,...,12.0,33.0,57.0,0.0,13.0,7.0,NaN,NaN,NaN,NaN
25,12/14/2025,Online,Heretics,Cloud9,Exposure,0,5,28,17,22,...,42.0,8.0,54.0,0.0,10.0,0.0,NaN,NaN,NaN,NaN
56,1/18/2026,Online,OpTic,Thieves,Scar,5,21,14,40,28,...,3.0,48.0,52.0,1.0,25.0,18.0,NaN,NaN,NaN,NaN
106,2/14/2026,Online,Surge,Thieves,Scar,11,6,48,0,6,...,11.0,30.0,46.0,1.0,32.0,17.0,0.0,49.0,3.0,21.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
69,1/25/2026,Online,Thieves,Ravens,Den,24,15,19,28,37,...,6.0,1.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
82,1/30/2026,LAN,FaZe,Breach,Den,24,22,41,10,11,...,28.0,19.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
102,2/14/2026,Online,Falcons,M8s,Scar,13,24,16,24,21,...,0.0,23.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
105,2/14/2026,Online,KOI,FaZe,Blackheart,11,29,22,31,1,...,13.0,28.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


Going to the February 13th match between we find that the raw value for "Hill 9 A" was incorrectly input as 149 instead of 249, leading to the oddity we found. Both of these will be fixed in our spreadsheet, then we will load the new one and perform the same transformations as we did above.

In [105]:
bo7raw2 = pd.read_csv('cdlHillCountsBO7v2.csv')
timePerHill2 = bo7raw2.copy()

timePerHill2['Hill 13 A'] -= timePerHill2['Hill 12 A']
timePerHill2['Hill 13 B'] -= timePerHill2['Hill 12 B']
timePerHill2['Hill 12 A'] -= timePerHill2['Hill 11 A']
timePerHill2['Hill 12 B'] -= timePerHill2['Hill 11 B']
timePerHill2['Hill 11 A'] -= timePerHill2['Hill 10 A']
timePerHill2['Hill 11 B'] -= timePerHill2['Hill 10 B']
timePerHill2['Hill 10 A'] -= timePerHill2['Hill 9 A']
timePerHill2['Hill 10 B'] -= timePerHill2['Hill 9 B']
timePerHill2['Hill 9 A'] -= timePerHill2['Hill 8 A']
timePerHill2['Hill 9 B'] -= timePerHill2['Hill 8 B']
timePerHill2['Hill 8 A'] -= timePerHill2['Hill 7 A']
timePerHill2['Hill 8 B'] -= timePerHill2['Hill 7 B']
timePerHill2['Hill 7 A'] -= timePerHill2['Hill 6 A']
timePerHill2['Hill 7 B'] -= timePerHill2['Hill 6 B']
timePerHill2['Hill 6 A'] -= timePerHill2['Hill 5 A']
timePerHill2['Hill 6 B'] -= timePerHill2['Hill 5 B']
timePerHill2['Hill 5 A'] -= timePerHill2['Hill 4 A']
timePerHill2['Hill 5 B'] -= timePerHill2['Hill 4 B']
timePerHill2['Hill 4 A'] -= timePerHill2['Hill 3 A']
timePerHill2['Hill 4 B'] -= timePerHill2['Hill 3 B']
timePerHill2['Hill 3 A'] -= timePerHill2['Hill 2 A']
timePerHill2['Hill 3 B'] -= timePerHill2['Hill 2 B']
timePerHill2['Hill 2 A'] -= timePerHill2['Hill 1 A']
timePerHill2['Hill 2 B'] -= timePerHill2['Hill 1 B']

timePerHill2[timePerHill2.columns[5:31]].max()

Hill 1 A     40.0
Hill 1 B     40.0
Hill 2 A     57.0
Hill 2 B     57.0
Hill 3 A     53.0
Hill 3 B     55.0
Hill 4 A     53.0
Hill 4 B     56.0
Hill 5 A     56.0
Hill 5 B     57.0
Hill 6 A     52.0
Hill 6 B     55.0
Hill 7 A     54.0
Hill 7 B     53.0
Hill 8 A     56.0
Hill 8 B     55.0
Hill 9 A     59.0
Hill 9 B     55.0
Hill 10 A    57.0
Hill 10 B    55.0
Hill 11 A    48.0
Hill 11 B    58.0
Hill 12 A    50.0
Hill 12 B    49.0
Hill 13 A    19.0
Hill 13 B    24.0
dtype: float64

In [106]:
timePerHill2[timePerHill2.columns[5:31]].min()

Hill 1 A      0.0
Hill 1 B      0.0
Hill 2 A      0.0
Hill 2 B      0.0
Hill 3 A      0.0
Hill 3 B      0.0
Hill 4 A      0.0
Hill 4 B      0.0
Hill 5 A      0.0
Hill 5 B      0.0
Hill 6 A      0.0
Hill 6 B      0.0
Hill 7 A      0.0
Hill 7 B      0.0
Hill 8 A      0.0
Hill 8 B      0.0
Hill 9 A    -65.0
Hill 9 B      0.0
Hill 10 A   -65.0
Hill 10 B     0.0
Hill 11 A   -80.0
Hill 11 B     0.0
Hill 12 A     0.0
Hill 12 B     0.0
Hill 13 A     0.0
Hill 13 B     0.0
dtype: float64

The maxes look good, but some of the same issues are occuring on the opposite end. The same checks will be done so that we can fix the mistakes and move on.

In [108]:
check9a = timePerHill2.sort_values(by=['Hill 9 A'])
check9a

,Date,Venue,Team A,Team B,Map,Hill 1 A,Hill 1 B,Hill 2 A,Hill 2 B,Hill 3 A,...,Hill 9 A,Hill 9 B,Hill 10 A,Hill 10 B,Hill 11 A,Hill 11 B,Hill 12 A,Hill 12 B,Hill 13 A,Hill 13 B
19,12/13/2025,Online,M8s,Thieves,Den,16,11,23,14,40,...,-65.0,0.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
53,1/18/2026,Online,G2,FaZe,Blackheart,16,15,20,31,43,...,0.0,48.0,2.0,25.0,NaN,NaN,NaN,NaN,NaN,NaN
58,1/23/2026,Online,OpTic,Ravens,Den,16,24,13,35,36,...,0.0,18.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
107,2/14/2026,Online,Surge,Thieves,Exposure,15,15,0,45,46,...,0.0,13.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
81,1/30/2026,LAN,Falcons,Cloud9,Den,34,7,55,0,6,...,0.0,39.0,21.0,16.0,5.0,21.0,NaN,NaN,NaN,NaN
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
4,12/6/2025,Online,KOI,Heretics,Scar,29,0,51,0,51,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
28,12/14/2025,Online,KOI,FaZe,Colossus,9,25,16,31,0,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
44,1/16/2026,Online,Falcons,M8s,Exposure,7,21,10,42,21,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
52,1/18/2026,Online,Cloud9,Breach,Colossus,4,27,53,0,25,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [109]:
check10a = timePerHill2.sort_values(by=['Hill 10 A'])
check10a

,Date,Venue,Team A,Team B,Map,Hill 1 A,Hill 1 B,Hill 2 A,Hill 2 B,Hill 3 A,...,Hill 9 A,Hill 9 B,Hill 10 A,Hill 10 B,Hill 11 A,Hill 11 B,Hill 12 A,Hill 12 B,Hill 13 A,Hill 13 B
57,1/23/2026,Online,OpTic,Ravens,Colossus,24,6,19,37,4,...,16.0,26.0,-65.0,1.0,NaN,NaN,NaN,NaN,NaN,NaN
0,12/5/2025,Online,Falcons,OpTic,Scar,0,40,24,19,18,...,20.0,29.0,0.0,5.0,NaN,NaN,NaN,NaN,NaN,NaN
140,2/28/2026,Online,Surge,Ravens,Blackheart,35,5,25,20,0,...,21.0,25.0,0.0,32.0,NaN,NaN,NaN,NaN,NaN,NaN
118,2/20/2026,Online,KOI,Thieves,Colossus,4,29,50,3,8,...,33.0,17.0,0.0,11.0,NaN,NaN,NaN,NaN,NaN,NaN
104,2/14/2026,Online,OpTic,G2,Scar,24,6,16,34,7,...,44.0,4.0,0.0,55.0,32.0,5.0,5.0,2.0,NaN,NaN
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
125,2/22/2026,Online,Falcons,Ravens,Blackheart,24,19,25,24,2,...,16.0,5.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
126,2/22/2026,Online,Surge,Breach,Blackheart,31,2,25,26,31,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
132,2/27/2026,Online,Ravens,Breach,Den,33,3,4,43,32,...,9.0,4.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
134,2/27/2026,Online,Falcons,FaZe,Blackheart,10,26,28,17,27,...,7.0,0.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [110]:
check11a = timePerHill2.sort_values(by=['Hill 11 A'])
check11a

,Date,Venue,Team A,Team B,Map,Hill 1 A,Hill 1 B,Hill 2 A,Hill 2 B,Hill 3 A,...,Hill 9 A,Hill 9 B,Hill 10 A,Hill 10 B,Hill 11 A,Hill 11 B,Hill 12 A,Hill 12 B,Hill 13 A,Hill 13 B
142,2/28/2026,Online,OpTic,Thieves,Colossus,6,27,28,23,11,...,26.0,9.0,2.0,41.0,-80.0,1.0,NaN,NaN,NaN,NaN
21,12/13/2025,Online,OpTic,Heretics,Den,16,26,20,34,41,...,18.0,31.0,22.0,29.0,0.0,39.0,NaN,NaN,NaN,NaN
43,1/16/2026,Online,Surge,Heretics,Exposure,13,8,32,15,6,...,32.0,15.0,27.0,26.0,0.0,8.0,NaN,NaN,NaN,NaN
137,2/28/2026,Online,Falcons,G2,Scar,30,0,50,0,7,...,26.0,11.0,9.0,37.0,0.0,29.0,24.0,12.0,19.0,5.0
8,12/6/2025,Online,G2,Cloud9,Scar,1,8,29,13,9,...,41.0,7.0,16.0,20.0,0.0,58.0,33.0,4.0,NaN,NaN
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
138,2/28/2026,Online,Falcons,G2,Colossus,6,18,46,1,19,...,5.0,27.0,12.0,22.0,NaN,NaN,NaN,NaN,NaN,NaN
140,2/28/2026,Online,Surge,Ravens,Blackheart,35,5,25,20,0,...,21.0,25.0,0.0,32.0,NaN,NaN,NaN,NaN,NaN,NaN
144,3/1/2026,Online,FaZe,Breach,Blackheart,5,25,44,1,20,...,13.0,0.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
146,3/1/2026,Online,Heretics,Cloud9,Scar,0,40,17,19,24,...,34.0,11.0,3.0,0.0,NaN,NaN,NaN,NaN,NaN,NaN


In [111]:
bo7raw3 = pd.read_csv('cdlHillCountsBO7v3.csv')
timePerHill3 = bo7raw3.copy()

timePerHill3['Hill 13 A'] -= timePerHill3['Hill 12 A']
timePerHill3['Hill 13 B'] -= timePerHill3['Hill 12 B']
timePerHill3['Hill 12 A'] -= timePerHill3['Hill 11 A']
timePerHill3['Hill 12 B'] -= timePerHill3['Hill 11 B']
timePerHill3['Hill 11 A'] -= timePerHill3['Hill 10 A']
timePerHill3['Hill 11 B'] -= timePerHill3['Hill 10 B']
timePerHill3['Hill 10 A'] -= timePerHill3['Hill 9 A']
timePerHill3['Hill 10 B'] -= timePerHill3['Hill 9 B']
timePerHill3['Hill 9 A'] -= timePerHill3['Hill 8 A']
timePerHill3['Hill 9 B'] -= timePerHill3['Hill 8 B']
timePerHill3['Hill 8 A'] -= timePerHill3['Hill 7 A']
timePerHill3['Hill 8 B'] -= timePerHill3['Hill 7 B']
timePerHill3['Hill 7 A'] -= timePerHill3['Hill 6 A']
timePerHill3['Hill 7 B'] -= timePerHill3['Hill 6 B']
timePerHill3['Hill 6 A'] -= timePerHill3['Hill 5 A']
timePerHill3['Hill 6 B'] -= timePerHill3['Hill 5 B']
timePerHill3['Hill 5 A'] -= timePerHill3['Hill 4 A']
timePerHill3['Hill 5 B'] -= timePerHill3['Hill 4 B']
timePerHill3['Hill 4 A'] -= timePerHill3['Hill 3 A']
timePerHill3['Hill 4 B'] -= timePerHill3['Hill 3 B']
timePerHill3['Hill 3 A'] -= timePerHill3['Hill 2 A']
timePerHill3['Hill 3 B'] -= timePerHill3['Hill 2 B']
timePerHill3['Hill 2 A'] -= timePerHill3['Hill 1 A']
timePerHill3['Hill 2 B'] -= timePerHill3['Hill 1 B']

timePerHill3[timePerHill3.columns[5:31]].min()

Hill 1 A     0.0
Hill 1 B     0.0
Hill 2 A     0.0
Hill 2 B     0.0
Hill 3 A     0.0
Hill 3 B     0.0
Hill 4 A     0.0
Hill 4 B     0.0
Hill 5 A     0.0
Hill 5 B     0.0
Hill 6 A     0.0
Hill 6 B     0.0
Hill 7 A     0.0
Hill 7 B     0.0
Hill 8 A     0.0
Hill 8 B     0.0
Hill 9 A     0.0
Hill 9 B     0.0
Hill 10 A    0.0
Hill 10 B    0.0
Hill 11 A    0.0
Hill 11 B    0.0
Hill 12 A    0.0
Hill 12 B    0.0
Hill 13 A    0.0
Hill 13 B    0.0
dtype: float64

Looks good! Now, based off of what we just looked at, the first point I want to explore is the amount of shutouts occuring for each hill. From that we can get a get an idea of what points tend to be the easiest to defend, and what points have the most back and forth fighting taking place.

In [113]:
shutouts = (timePerHill3[timePerHill3.columns[5:31]] == 0).sum()
shutouts

Hill 1 A     10
Hill 1 B      6
Hill 2 A     19
Hill 2 B     17
Hill 3 A      9
Hill 3 B      4
Hill 4 A      9
Hill 4 B      7
Hill 5 A      6
Hill 5 B      7
Hill 6 A     10
Hill 6 B      6
Hill 7 A     11
Hill 7 B      7
Hill 8 A      9
Hill 8 B      8
Hill 9 A      6
Hill 9 B      7
Hill 10 A    11
Hill 10 B    11
Hill 11 A     8
Hill 11 B    11
Hill 12 A     9
Hill 12 B    11
Hill 13 A     2
Hill 13 B     1
dtype: int64

Ok, so here we get an idea of how many shutouts are hapenning in general, but we can't learn much from this. What would be better is if we grouped them by map. Then, knowing each map only has 4 or 5 actual hill locations, we can sum each unique hill with the others that ocurred at the same location.

In [115]:
mapLook = timePerHill3[timePerHill3.columns[4:31]]
mapLook.iloc[:, 1:27] = mapLook.iloc[:, 1:27].apply(lambda x: x==0)
mapLook = mapLook.groupby('Map')[mapLook.columns[1:27]].sum().reset_index()

fourHillMaps = ['Colossus', 'Den']
fiveHillMaps = ['Scar', 'Exposure', 'Blackheart']

hps4 = mapLook[mapLook['Map'].isin(fourHillMaps)].copy()
hps5 = mapLook[mapLook['Map'].isin(fiveHillMaps)].copy()

hps4['P1'] = hps4['Hill 1 A']+hps4['Hill 1 B'] + hps4['Hill 5 A']+hps4['Hill 5 B'] + hps4['Hill 9 A']+hps4['Hill 9 B'] + hps4['Hill 13 A']+hps4['Hill 13 B']
hps4['P2'] = hps4['Hill 2 A']+hps4['Hill 2 B'] + hps4['Hill 6 A']+hps4['Hill 6 B'] + hps4['Hill 10 A']+hps4['Hill 10 B']
hps4['P3'] = hps4['Hill 3 A']+hps4['Hill 3 B'] + hps4['Hill 7 A']+hps4['Hill 7 B'] + hps4['Hill 11 A']+hps4['Hill 11 B']
hps4['P4'] = hps4['Hill 4 A']+hps4['Hill 4 B'] + hps4['Hill 8 A']+hps4['Hill 8 B'] + hps4['Hill 12 A']+hps4['Hill 12 B']

hps5['P1'] = hps5['Hill 1 A']+hps5['Hill 1 B'] + hps5['Hill 6 A']+hps5['Hill 6 B'] + hps5['Hill 11 A']+hps5['Hill 11 B']
hps5['P2'] = hps5['Hill 2 A']+hps5['Hill 2 B'] + hps5['Hill 7 A']+hps5['Hill 7 B'] + hps5['Hill 12 A']+hps5['Hill 12 B']
hps5['P3'] = hps5['Hill 3 A']+hps5['Hill 3 B'] + hps5['Hill 8 A']+hps5['Hill 8 B'] + hps5['Hill 13 A']+hps5['Hill 13 B']
hps5['P4'] = hps5['Hill 4 A']+hps5['Hill 4 B'] + hps5['Hill 9 A']+hps5['Hill 9 B'] 
hps5['P5'] = hps5['Hill 5 A']+hps5['Hill 5 B'] + hps5['Hill 10 A']+hps5['Hill 10 B']

hps4 = hps4[['Map', 'P1', 'P2', 'P3', 'P4']]
hps5 = hps5[['Map', 'P1', 'P2', 'P3', 'P4', 'P5']]
shutoutsByMap = pd.concat([hps4, hps5])
shutoutsByMap

,Map,P1,P2,P3,P4,P5
1,Colossus,2,18,10,14,NaN
2,Den,10,17,8,10,NaN
0,Blackheart,5,14,8,6,6.0
3,Exposure,14,9,8,5,7.0
4,Scar,14,19,6,3,9.0


There it is! Now we can get an idea of which hills are the "mixiest", like P1 Colossus, or P4 Scar. The biggest "money hills" appear to be P2s on Colossus, Scar, and Blackheart.

Though some of these findings aren't too surprising for keen observers of the CDL, we can use the same knowledge to question some of our findings as well. For instance, the first hill, "P1", is often set up to be an even battle for each team as it is the first of the game. This leads to scenarios where some teams avoid scoring points in favor of attempting to gain good positioning. This could be boosting the amount of times where shutouts occur.

It is also known that the map "Blackheart" was not able to be played for the beginning portion of the season due to bugs. This likely means that the "money hills" on this map are even more important given there was less recorded matches played there in comparison to others.

These ideas will be further explored, in addition to viewing team performance in this context.